### PR TITLE
Add project index to repo-level readme

### PR DESCRIPTION
**Pull request by tokenstar**
Add project index to repo-level readme and update PR template to refer to it

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -12,10 +12,10 @@ Please delete options that are not relevant.
 
 # Checklist:
 
-- [ ] My content uses the appropriate readme template
+- [ ] My content uses the appropriate readme [template](https://github.com/Azure/azure-sphere-gallery/tree/main/Templates)
 - [ ] I have performed a self-review of my own contribution
 - [ ] I have provided comments where appropriate
-- [ ] I have made corresponding changes to the README.md and other relevant doc files
-- [ ] I have made the appropriate licensing updates
+- [ ] I have updated the repository's [README.md](https://github.com/Azure/azure-sphere-gallery/blob/main/README.md) to index this project
```
Comment (cawhitworth) [inline .github/pull_request_template.md:18]: Do we want to keep the "and any other relevant documents"? Perhaps as a separate bullet/check to say "No other doc updates needed" or list out other updates?

**Review by cawhitworth — APPROVED** [review]

File: .github/pull_request_template.md (hunk 1 above)
Comment (tokenstar) [reply .github/pull_request_template.md:18]: after discussion, not doing this - so that the checklist is specific/actionable

**Review by rebeccaleigh — APPROVED** [review]

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -12,10 +12,10 @@ Please delete options that are not relevant.
...
+- [ ] I have added/updated licence(s) for this project as appropriate
```
Comment (pennyo) [inline .github/pull_request_template.md:19]: license (American English is MS standard)

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -12,10 +12,10 @@ Please delete options that are not relevant.
 
 # Checklist:
 
-- [ ] My content uses the appropriate readme template
+- [ ] My content uses the appropriate readme [template](https://github.com/Azure/azure-sphere-gallery/tree/main/Templates)
```
Comment (pennyo) [inline .github/pull_request_template.md:15]: Can we add a line to the effect that my content contains a single README.md file that is based on the template(s) and is named README.md?

I know it sounds stupid, but you'd be surprised how many people don't rename a template -- or clean up the contents--when they use one.

**Review by pennyo — APPROVED** [review]

**Pull request merged**